pkgs/tools/admin/mycli/default.nix

```diff
@@ -1,28 +1,31 @@
 { lib
 , python
+, pythonPackages
 }:
 
 with python.pkgs;
 
 buildPythonApplication rec {
   pname = "mycli";
-  version = "1.6.0";
+  version = "1.17.0";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "0qg4b62kizyb16kk0cvpk70bfs3gg4q4hj2b15nnc7a3gqqfp67j";
+    sha256 = "11d3ssjifms6bid77jk06zl5wl3srihijmv5kggxa0w2l59y8h9m";
   };
 
   propagatedBuildInputs = [
-    pymysql configobj sqlparse prompt_toolkit pygments click pycrypto
+    pymysql configobj sqlparse prompt_toolkit pygments click pycrypto cli-helpers
   ];
 
-  postPatch = ''
-    substituteInPlace setup.py --replace "==" ">="
-  '';
+  patches = [ ./fix-tests.patch ];
+
+  checkInputs = with pythonPackages; [ pytest mock ];
 
-  # No tests in archive. Newer versions do include tests
-  doCheck = false;
+  preCheck = ''
+    echo '[main]' > /tmp/myclirc
+    echo 'log_file = /tmp/mycli.test.log' >> /tmp/myclirc
```

Comment:
From where would `myclirc` be read without the patch?

Reply:
From nowhere, and so the library will end up calling [write_default_config](https://github.com/dbcli/mycli/blob/dde8320b04f42345a1b83520ac646b2bc06ea229/mycli/main.py#L134-L135) which writes the config file at `/homeless-shelter/.myclirc` and that of course fails. Setting the myclirc stops it from writing the config file. Also, I specified the `log_file` option because another test invokes logging and that defaults to `/homeless-shelter/.mycli.log`.

Reply:
How about setting something like `HOME=.`?

Reply:
I'll try that!

Reply:
This worked! Thanks for the tip.